[GC-4654] Build a system in which we can easily install the new Content Workflow plugin

CWDN opened the pull request:
As we are looking to migrate from the old GatherContent wordpress plugin to the new Content Workflow one we need a way to make the migration process super easy. 

So what this does is add a little banner in the admin and gives the ability to install the plugin and activate it from the banner. This should hopefully reduce the friction to get the new plugin with all the new features.

We can definitely do more work in the banner to make it more appealing to do the migration.

Video demo:

https://share.zight.com/Bluvkq29

CWDN committed:
✨ Added the ability to install our new Content Workflow plugin from this plugin

Gallmond commented:
Makes sense to me.

NB: all those `gc_` will need to change to `cwby_` else WP folks will bounce it back I think.

CWDN committed:
👔 Added logic to now deactivate the old plugin

This is because the old and new plugin can't run at the same time so we
deactivate the old one.

CWDN committed:
✏️ Fixed the old plugin name being wrong

CWDN committed:
👔 Added handling to stop the old plugin activating while the new one is active
